flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java

```diff
@@ -125,6 +125,10 @@ public void testDisabledCheckpointing() throws Exception {
 	 */
 	@Test
 	public void testChainStartEndSetting() throws Exception {
+
+		// set parallelism to 4 to avoid chaining with source in case when available processors is 1.
+		final int parallelism = 4;
+
 		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
```

Comment:
I think it is easier to set `env.setParallelism(2)` instead of individually setting the parallelism on every operator.